### PR TITLE
DM-28090: CSC now returns CMD_INPROGESS when enabled or disabled.

### PR DESCRIPTION
**Pull request by wvreeven**

### CODE REVIEW

File: python/lsst/ts/weatherstation/csc.py [hunk 1]
```diff
@@ -118,6 +118,7 @@ async def end_enable(self, id_data):
         self.telemetry_loop_task = asyncio.ensure_future(self.telemetry_loop())
 
         await super().end_enable(id_data)
+        return SalRetCode.CMD_INPROGRESS
```
Comment (tribeiro) [inline python/lsst/ts/weatherstation/csc.py]: I should probably have been more clear.  The way to send an in progress ack is documented in https://ts-salobj.lsst.io/salobj_cscs.html#writing-a-csc 

In this case you would do:

```
self.cmd_enable.ack_in_progress(data, timeout=self.enable_timeout)
```

you probably want to do this is `begin_enable` though.

File: python/lsst/ts/weatherstation/csc.py [hunk 2]
```diff
@@ -144,6 +145,7 @@ async def begin_disable(self, id_data):
             self.fault()
 
         await super().begin_disable(id_data)
+        return SalRetCode.CMD_INPROGRESS
```
Comment (tribeiro) [inline python/lsst/ts/weatherstation/csc.py]: See my comment above about how to send the ack

File: python/lsst/ts/weatherstation/csc.py [hunk 3]
```diff
@@ -157,6 +159,7 @@ async def end_disable(self, id_data):
         await self.wait_loop(self.telemetry_loop_task)
 
         await super().end_disable(id_data)
+        return SalRetCode.CMD_INPROGRESS
```
Comment (tribeiro) [inline python/lsst/ts/weatherstation/csc.py]: Ack is already been sent in `begin_disable` no need to resend it here.

**Review by tribeiro — CHANGES_REQUESTED** [review]
I should probably have pointed you to the documentation or be more explicit about what needed to be done. I commented inline about how to send the ack.

File: python/lsst/ts/weatherstation/csc.py (hunk 1 above)
Comment (wvreeven) [reply python/lsst/ts/weatherstation/csc.py]: Thanks for the clarification. I adjusted the code accordingly.

File: python/lsst/ts/weatherstation/csc.py (hunk 2 above)
Comment (wvreeven) [reply python/lsst/ts/weatherstation/csc.py]: Thanks for the clarification. I adjusted the code accordingly.

File: python/lsst/ts/weatherstation/csc.py (hunk 3 above)
Comment (wvreeven) [reply python/lsst/ts/weatherstation/csc.py]: Thanks for the clarification. I adjusted the code accordingly.

**Comment by wvreeven** [comment]
Changes applied. Please check again. Many thanks!

**Comment by tribeiro** [comment]
You are sending the ack in the proper way now but you missed sending it in `begin_enable`

**Comment by wvreeven** [comment]
That was quite sloppy of me, sorry. I added the do_enable method now as well.

**Review by tribeiro — APPROVED** [review]

**Pull request merged**